samples/transform_feedback_separated.html

```diff
@@ -0,0 +1,237 @@
+</head>
+
+<body>
+    <div id="info">WebGL 2 Samples - transform_feedback_separated</div>
+
+    <script id="vs-transform" type="x-shader/x-vertex">
+        #version 300 es
+        #define POSITION_LOCATION 0
+
+        precision highp float;
+        precision highp int;
+
+        uniform mat4 MVP;
+        layout(location = POSITION_LOCATION) in vec4 position;
+
+        out vec4 color;
+
+        void main()
+        {
+            gl_Position = MVP * position;
+            color = vec4(clamp(vec2(position), 0.0, 1.0), 0.0, 1.0);
+        }
+    </script>
+
+    <!-- This fragment shader will not be called since it is used when the rasterizer is disabled. -->
+    <script id="fs-transform" type="x-shader/x-fragment">
+        #version 300 es
+        precision highp float;
+        precision highp int;
+
+        in vec4 color;
+        out vec4 OColor;
+
+        void main()
+        {
+            OColor = color;
+        }
+    </script>
+
+    <script id="vs-feedback" type="x-shader/x-vertex">
+        #version 300 es
+        #define POSITION_LOCATION 0
+        #define COLOR_POSITION 3
+
+        precision highp float;
+        precision highp int;
+
+        layout(location = POSITION_LOCATION) in vec4 position;
+        layout(location = COLOR_POSITION) in vec4 color;
+
+        out vec4 vColor;
+
+        void main()
+        {
+            gl_Position = position;
+            vColor = color;
+        }
+    </script>
+
+    <script id="fs-feedback" type="x-shader/x-fragment">
+        #version 300 es
+        precision highp float;
+        precision highp int;
+
+        in vec4 vColor;
+
+        out vec4 color;
+
+        void main()
+        {
+            color = vColor;
+        }
+    </script>
+
+    <script src="utility.js"></script>
+    <script>
+    (function () {
+        'use strict';
+
+        // -- Init Canvas
+        var canvas = document.createElement('canvas');
+        canvas.width = Math.min(window.innerWidth, window.innerHeight);
+        canvas.height = canvas.width;
+        document.body.appendChild(canvas);
+
+        // -- Init WebGL Context
+        var gl = canvas.getContext('webgl2', { antialias: false });
+        if (!gl) {
+            gl = canvas.getContext('experimental-webgl2', { antialias: false });
+        }
+        var isWebGL2 = !!gl;
+        if(!isWebGL2)
+        {
+            document.getElementById('info').innerHTML = 'WebGL 2 is not available.  See <a href="https://www.khronos.org/webgl/wiki/Getting_a_WebGL_Implementation">How to get a WebGL 2 implementation</a>';
+            return;
+        }
+
+        // -- Init Program
+        var programTransform = createProgram(gl, getShaderSource('vs-transform'), getShaderSource('fs-transform'));
+
+        var varyings = ['gl_Position', 'color'];
+        gl.transformFeedbackVaryings(programTransform, varyings, gl.SEPARATE_ATTRIBS);
+        gl.linkProgram(programTransform); // Relink program for transform feedback varyings
```

Comment:
Instead of relinking here, we should include all the required code from `createProgram` inline here.  Otherwise, this is too hard to follow.  See `transform_feedback_interleaved`.